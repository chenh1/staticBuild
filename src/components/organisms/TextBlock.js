import React, {PropTypes} from 'react';
import { Paragraph, SubHeader } from '../atoms';
import BulletList from '../molecules/BulletList';

const TextBlock = ({elements}) => {
    return (
        <div className="text-wrapper">
            {elements.map((element, index) => {
                switch (element.type) {
                    case 'paragraph':
                        return (
                            <Paragraph key={"text" + index} text={element.text}/>
                        );
                    case 'subHeader':
                        return (
                            <SubHeader key={"text" + index} customClass={element.customClass} text={element.text}/>
                        );
                    case 'bulletList':
                        return (
                            <BulletList key={"text" + index} listItems={element.listItems}/>
                        );
                    default:
                        return;
                }
            })}
        </div>
    );
};

TextBlock.propTypes = {
    elements: PropTypes.array.isRequired
};

export default TextBlock;