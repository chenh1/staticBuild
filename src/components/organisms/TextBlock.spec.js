import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import TextBlock from './TextBlock';

test('renders text block which conditionally renders different component types', () => {
    const tree = renderer.create(<TextBlock
        elements={[
            {type: 'paragraph', text: ['Mock paragraph text', '...more text']},
            {type: 'subHeader', text: 'Mock subheader text'},
            {type: 'bulletList', listItems: ['list item 1', 'list item 2']}
        ]} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});