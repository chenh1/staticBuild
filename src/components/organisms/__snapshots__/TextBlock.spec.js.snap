// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text block which conditionally renders different component types 1`] = `
<div
  className="text-wrapper"
>
  <p
    className="paragraph-one"
  >
    Mock paragraph text
    ...more text
  </p>
  <h3
    className="header-three"
  >
    Mock subheader text
  </h3>
  <ul
    className="list-one"
  >
    <li>
      <span>
        list item 1
      </span>
    </li>
    <li>
      <span>
        list item 2
      </span>
    </li>
  </ul>
</div>
`;
